app/src/main/java/org/oppia/app/topic/TopicActivity.kt

```diff
@@ -18,13 +18,18 @@ const val TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY = "TopicActivity.topic_id"
 class TopicActivity : InjectableAppCompatActivity(), RouteToQuestionPlayerListener, RouteToConceptCardListener,
   RouteToTopicPlayListener, RouteToStoryListener, RouteToExplorationListener {
   private lateinit var topicId: String
+  private lateinit var storyId: String
   @Inject
   lateinit var topicActivityPresenter: TopicActivityPresenter
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     activityComponent.inject(this)
     topicId = intent?.getStringExtra(org.oppia.app.topic.TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY) ?: TEST_TOPIC_ID_0
+    storyId = checkNotNull(
```

Comment:
This shouldn't be a checkNotNull because storyId actually is optional. storyId should be of type String? and should be null if the story ID is missing (in which case we don't auto scroll). I think that should help simplify downstream logic.

Reply:
Resolved